ostd/src/mm/io.rs

```diff
@@ -553,11 +582,40 @@ impl<'a> VmWriter<'a, KernelSpace> {
         Ok(())
     }
 
+    /// Writes a value of the `Pod` type using one non-tearing memory store.
+    ///
+    /// If the length of the `Pod` type exceeds `self.remain()`, this method will return `Err`.
+    ///
+    /// # Panics
+    ///
+    /// This method will not compile (due to failed constant assertions) if the `Pod` type is too
+    /// large for the current architecture and the operation must be tear into multiple memory
+    /// stores.
+    ///
+    /// This method will panic if the current position of the writer does not meet the alignment
+    /// requirements of type `T`.
+    pub fn write_val_no_tearing<T: Pod>(&mut self, new_val: &T) -> Result<()> {
+        const { assert_memory_access_no_tearing::<T>() };
+
+        if self.avail() < core::mem::size_of::<T>() {
+            return Err(Error::InvalidArgs);
+        }
+
+        assert!(self.cursor.cast::<T>().is_aligned());
```

Comment:
The pointer has been constructed twice. Once here, the other time just before `write_volatile`. You may want to consider adding a temp variable.